tests/core/anthropic/test_tool.py

```diff
@@ -18,7 +18,7 @@ class FormatBook(AnthropicTool):
         title: str
         author: str
 
-        tool_config = ToolConfig(cache_control={"type": "ephemeral"})
+        tool_config = AnthropicToolConfig(cache_control={"type": "ephemeral"})
```

Comment:
It may be difficult to do this with a type checker, but it is better to check at runtime level whether the Tool is compatible with the tool_config passed.

Reply:
This is difficult because the user will define a `BaseTool`, which will need to accept any `ToolConfig` subclass.

We could add runtime warnings or errors to all other providers, but I'm not sure how worthwhile that is since it won't break anything (it will just silently not use the feature unless the provider supports it).

Given the explicit naming of the ToolConfig to e.g. `AnthropicToolConfig`, I think it's clear enough for now, and we can revisit later if necessary.